Libs/MRML/Core/vtkMRMLStorageNode.h

```diff
@@ -26,6 +27,76 @@ class vtkStringArray;
 
 // STD includes
 #include <vector>
+#include <algorithm>
+
+///
+/// A class for recording warnings and errors associated with this
+/// vtkMRMLStorableNode.  This cannot be hidden within the
+/// vtkMRMLStorableNode class scope because the Python wrapping code
+/// vtkMRMLStorageNodePython.cxx cannot wrap the GetVTKMessageLog
+/// method.
+class vtkMessageLog
+{
+public:
+  using event_type = unsigned long;
+  ///
+  /// A class that describes a single message
+  struct Message
+  {
+    Message(event_type eventId, const std::string &messageText)
+      : EventId(eventId), MessageText(messageText) {}
+    ///
+    /// Function object for checking whether a Message has a specified
+    /// EventID.
+    struct EqualEventId
+    {
+      EqualEventId(event_type eventId) : EventId(eventId) {}
+      bool operator()(const Message &message) const
+        { return message.EventId == this->EventId; }
+
+      event_type EventId;
```

Comment:
Since you are using these with values like `vtkCommand::ErrorEvent` they aren't really `EventId`s as much as `MessageType`s.  `MessageType` would make a better field name.

Also maybe this could be implemented as a subclass of a `vtkVariantArray` so that you wouldn't need to invent a new API for the common operations.

Reply:
I will change EventId to MessageType.  I will take a look at `vtkVariantArray`.

Reply:
The `vtkVariantArray` discussion is continued below.  Marking this thread as resolved.